src/components/PropertyCard.tsx

```diff
@@ -0,0 +1,37 @@
+import { Link } from "@tanstack/react-router";
+import NiceToHaveFeatures from "./NiceToHaveFeatures";
+import { PropertyAllDetails } from "./PropertyDetails";
+
+function PropertyCard({
+  zpid,
+  imgSrc,
+  price,
+  propertyDetails,
+  bedrooms,
+  bathrooms,
+  streetAddress,
+  city,
+}: PropertyAllDetails) {
+  return (
+    <Link to="/details/$propertyId" params={{ propertyId: zpid }}>
+      <img
+        src={imgSrc}
+        alt="Property Image"
```

Comment:
Its  a good idea to keep components in its own folder. You can undo this change.